__tests__/endpoint.tests.js

```diff
@@ -103,6 +103,22 @@ describe('GET/api/articles', () => {
         expect(result.body.articles).toBeSortedBy('created_at', {descending:true});
         expect(Object.keys(result.body.articles[0])).toEqual(expectedColumns);
     })
+    it('200: accepts a topic query which filters articles by topic', async () => {
+        const result = await request(app).get('/api/articles?topic=cats');
+        const expectedColumns = ['article_id', 'title', 'topic', 'author', 'created_at', 'votes', 'article_img_url', 'comment_count'];
+
+        expect(result.status).toBe(200);
+        expect(result.body.articles).toHaveLength(1);
+        expect(result.body.articles).toBeSortedBy('created_at', {descending:true});
+        expect(Object.keys(result.body.articles[0])).toEqual(expectedColumns);
+    })
+    it('404: returns not found if given a topic which no articles use', async () => {
+        const result = await request(app).get('/api/articles?topic=notatopic');
+
+        expect(result.status).toBe(404);
+        const err = result.body;
+        expect(err.msg).toBe('No articles for topic: notatopic');
+    })
```

Comment:
Not a topic 👍 
No articles or this topic shouldn't throw an error if the topic exits.

Thanks :)